### PR TITLE
refactor: GetFirstStoredBlock() and getblockchaininfo follow-ups

### PR DESCRIPTION
**Pull request by jonatack**
Picks up the remaining review feedback in #21726 and #24956.

- make the global function `GetFirstStoredBlock()` a member of the `BlockManager` class
- pass the `start_block` param of `GetFirstStoredBlock()` by reference instead of a pointer
- use `GetBlockTime()` for RPC getblockchaininfo#time

### CODE REVIEW

**Comment by jonatack** [comment]
Updated now that #24956 is merged, and ready for review.

**Review by maflcko — APPROVED** [review]
ACK e2b954e87f0c4cd5c8ac6e4d9c6b4d784844b4d2

File: src/node/blockstorage.cpp [hunk 1]
```diff
@@ -390,10 +390,10 @@ bool BlockManager::IsBlockPruned(const CBlockIndex* pblockindex)
     return (m_have_pruned && !(pblockindex->nStatus & BLOCK_HAVE_DATA) && pblockindex->nTx > 0);
 }
 
-const CBlockIndex* GetFirstStoredBlock(const CBlockIndex* start_block) {
+const CBlockIndex* BlockManager::GetFirstStoredBlock(const CBlockIndex& start_block)
```
Comment (MarcoFalke) [inline src/node/blockstorage.cpp:393]: In theory `start_block` could be marked LIFETIMEBOUND.

Comment (jonatack) [reply src/node/blockstorage.cpp:393]: > In theory `start_block` could be marked LIFETIMEBOUND.

Ah, yes; done in #25060.

**Pull request merged**